### PR TITLE
refactor: switch Int and Uint to cosmossdk.io/math

### PR DESCRIPTION
**Pull request by p0mvn**
Progress towards: https://github.com/osmosis-labs/osmosis/issues/6099

## What is the purpose of the change

Moving `Int` and `Uint` to sdkmath. The changes are based on: https://github.com/cosmos/cosmos-sdk/pull/11788

## Brief Changelog

- type alias `Int` and `Uint` to cosmossdk.io/math
- Delete in-repo implementations
- Replace instances of missing `ToDec` API with `NewDecFromInt`

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Review by PaddyMc — APPROVED** [review]
LGTM :muscle:

File: types/coin.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func (coin Coin) IsNegative() bool {
 
 // IsNil returns true if the coin amount is nil and false otherwise.
 func (coin Coin) IsNil() bool {
-	return coin.Amount.i == nil
+	return coin.Amount.BigInt() == nil
```
Comment (mattverse) [inline types/coin.go:149]: This got me confusing at first, I think we should have something better cosmossdk.io/math side long term 😭

File: types/decimal.go [hunk 2]
```diff
@@ -331,7 +333,7 @@ func (d Dec) MulInt(i Int) Dec {
 }
 
 func (d Dec) MulIntMut(i Int) Dec {
-	d.i.Mul(d.i, i.i)
+	d.i.Mul(d.i, i.BigInt())
```
Comment (mattverse) [inline types/decimal.go:336]: What happens if `i Int` is nil here?

**Review by mattverse — APPROVED** [review]
Confirmed all instances of sdk.Dec has been removed lgtm

File: types/decimal.go (hunk 2 above)
Comment (p0mvn) [reply types/decimal.go:336]: Nothing changes compared to the current design, I believe this is an API only change

**Pull request merged**